### PR TITLE
[Serverless] serverless-init for containers

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

Init commit for supporting serverless container instrumentation

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/serverless-init/initcontainer/initcontainer.go
Comment (DarcyRaynerDD) [inline cmd/serverless-init/initcontainer/initcontainer.go:67]: Just so I understand the flow here, we wait for the process to complete, then we do one final flush if we had some kind of failure? What if err == nil? Why wouldn't we do a final flush then as well?

Comment (DarcyRaynerDD) [reply cmd/serverless-init/initcontainer/initcontainer.go:67]: Also, might make sense to have flushing be it's own function

File: cmd/serverless-init/initcontainer/initcontainer.go
Comment (DarcyRaynerDD) [inline cmd/serverless-init/initcontainer/initcontainer.go:1]: We might want to exclude these files from the windows compile, since they rely on unix process group ids.

File: cmd/serverless-init/initcontainer/initcontainer.go
Comment (DarcyRaynerDD) [inline cmd/serverless-init/initcontainer/initcontainer.go:32]: Will the ServerlessMetricsAgent and ServerlessTraceAgent, logs periodically flush?  I thought we disabled that behaviour.

File: cmd/serverless-init/metric/metric.go
Comment (DarcyRaynerDD) [inline cmd/serverless-init/metric/metric.go:14]: metric.Shutdown seems like an ambigous name, it wouldn't immediately jump into my mind that it's creating an shutdown enhanced metric.

File: cmd/serverless-init/main.go
Comment (DarcyRaynerDD) [inline cmd/serverless-init/main.go:24]: Is it possible for a user to misconfigure their setup, and see this error message? Should it be more descriptive?

**Review by DarcyRaynerDD — CHANGES_REQUESTED** [review]

File: cmd/serverless-init/initcontainer/initcontainer.go
Comment (maxday) [reply cmd/serverless-init/initcontainer/initcontainer.go]: flushing is now in a separate method + final flush is done no matter what

File: cmd/serverless-init/initcontainer/initcontainer.go
Comment (maxday) [reply cmd/serverless-init/initcontainer/initcontainer.go:41]: - cold start is metric is flushed right after being emitted
- all other metrics are flushed when the container is killed
- traces are flushed every 3s for now

we could change this behaviour of course but we might want to try that for the private beta?

Comment (DarcyRaynerDD) [reply cmd/serverless-init/initcontainer/initcontainer.go:41]: So any custom metrics might not be flushed for a long time?

**Review by DarcyRaynerDD — APPROVED** [review]

File: cmd/serverless-init/log/log.go
Comment (sgnn7) [inline cmd/serverless-init/log/log.go:66]: nit: Not much is saved in this block by joining the `if` and the condition

File: cmd/serverless-init/log/log.go
Comment (sgnn7) [inline cmd/serverless-init/log/log.go:72]:
```suggestion
			log.Errorf("Unable to change the log level: %s", err)
```

**Review by sgnn7 — APPROVED** [review]
@maxday Couple of tiny nits but lgtm. 

PS: I'm guessing that these new paths also have a corresponding change in CODEOWNERS too

**Review by sgnn7 — APPROVED** [review]
LGTM (assuming that `cmd/serverless-init` will be reviewed by @DataDog/serverless)

**Pull request merged**
